src/machines/commerce-machine.ts

```diff
@@ -182,9 +183,27 @@ export const commerceMachine = createMachine<
             return {}
           }
 
+          const result = z
+            .object({plans: z.array(pricingPlanSchema)})
+            .safeParse(context.pricingData)
+          if (!result.success) return {pricingData: event.data}
```

Comment:
When the machine first boots, it is possible for `context.pricingData` to be `{}`. This bit of Zod `safeParse` gives us confidence that if we don't early return, then we have access to a `plans` array of `PricingPlan` objects.